source/MaterialXView/Editor.cpp

```diff
@@ -21,6 +24,55 @@ class EditorFormHelper : public ng::FormHelper
     void setVariableSpacing(int val) { mVariableSpacing = val; }
 };
 
+// Custom color picker so we can get numeric entry and feedback.
+//
+class MyColorPicker : public ng::ColorPicker
+{
+public:
+    MyColorPicker(ng::Widget *parent, const ng::Color& color) :
+        ng::ColorPicker(parent, color)
+    {
+        ng::Popup *popup = this->popup();
+        ng::Widget *floatGroup = new ng::Widget(popup);
+        auto layout =
+            new ng::GridLayout(ng::Orientation::Horizontal, 2,
+                ng::Alignment::Middle, 2, 2);
+        layout->setColAlignment({ ng::Alignment::Fill, ng::Alignment::Fill });
+        layout->setSpacing(1, 1);
+        floatGroup->setLayout(layout);
+
+        const std::vector<std::string> colorLabels = { "Red", "Green", "Blue", "Alpha" };
+        for (size_t i = 0; i < colorLabels.size(); i++)
+        {
+            new ng::Label(floatGroup, colorLabels[i]);
+            mColorWidgets[i] = new ng::FloatBox<float>(floatGroup, color[i]);
+            mColorWidgets[i]->setEditable(true);
+            mColorWidgets[i]->setAlignment(ng::TextBox::Alignment::Right);
+            mColorWidgets[i]->setFixedSize(ng::Vector2i(70, 20));
+            mColorWidgets[i]->setFontSize(15);
+            mColorWidgets[i]->setSpinnable(true);
+            mColorWidgets[i]->setCallback([&](float)
+            {
+                ng::Color value(mColorWidgets[0]->value(), mColorWidgets[1]->value(), mColorWidgets[2]->value(), mColorWidgets[3]->value());
+                mColorWheel->setColor(value);
+                mPickButton->setBackgroundColor(value);
+                mPickButton->setTextColor(value.contrastingColor());
+            });
+        }
+
+        // The color wheel does not handle alpha properly, so only
+        // overwrite RGB in the callback.
+        mCallback = [&](const ng::Color &value) {
+            mColorWidgets[0]->setValue(value[0]);
+            mColorWidgets[1]->setValue(value[1]);
+            mColorWidgets[2]->setValue(value[2]);
+        };
+    }
+
+    // Additional numeric entry / feedback widgets
+    ng::FloatBox<float>* mColorWidgets[4];
```

Comment:
Nitpick: different code convention using `mFoo` instead of our normal `_foo`

Reply:
I'll change it. I was using the nano-gui convention but should use ours.